Improved directive handling for places that use them

bbakerman opened the pull request:
#518 Improved directive handling for places that use them

andimarek commented on src/main/java/graphql/language/Directive.java:
```diff
@@ -25,14 +30,38 @@ public Directive(String name, List<Argument> arguments) {
 
-    public void setName(String name) {
-        this.name = name;
+    /*
```
Can we separate this helper from the actual class? For examples in a class `Directives` following JDK Conventions.

bbakerman replied:
changed

andimarek commented on src/main/java/graphql/language/TypeDefinition.java:
```diff
@@ -14,4 +15,18 @@
      */
     List<Directive> getDirectives();
 
+    /**
```
I am not sure about this: These methods do not add any real new value to the interface: they are just convenient.

If we really need/want an Interface with these methods we should add something like `DirectivesContainer` and make all classes implement is directly instead via `TypeDefinition`.

bbakerman replied:
removed

bbakerman commented:
Ok I renamed the method, removed the TypeDefinition part.

I didnt create a Directives (plural) since there already was one.  I know packages could work.  Instead I created a NodesUtil helper

andimarek commented on src/main/java/graphql/language/NodeUtil.java:
```diff
@@ -0,0 +1,38 @@
+        return getByName(directives, Directive::getName);
+    }
+
+    public static Map<String, Argument> argumentByName(List<Argument> arguments) {
```
just a tiny things: `argumentsByName` (plural) :-)
